### PR TITLE
ENH: add save peaks to dipy_fit_dti, dki

### PR DESCRIPTION
**Pull request by alexrockhill**
Fixes https://github.com/dipy/dipy/issues/3122

Since I was there, I also added saving the peaks to the dki workflow and the generalized FA value to the dsi workflow for standardization.

Maybe you'd have a minute to review @skoudoro?

### CODE REVIEW

**Comment by skoudoro** [comment]
Thank you for this @alexrockhill, 

Overall, it looks good, but I need to try that locally.  I will come back to you it bit later

**Comment by alexrockhill** [comment]
Sounds good, take your time

**Comment by alexrockhill** [comment]
Looks like an unrelated checksum error but I'm not able to reproduce it locally, I get the checksum on `master`.

File: dipy/workflows/reconst.py [hunk 1]
```diff
@@ -305,6 +311,28 @@ def run(self, input_files, bvalues_files, bvectors_files, mask_files,
...
+        out_shm : string, optional
+            Name of the spherical harmonics volume to be saved.
+        out_peaks_dir : string, optional
+            Name of the peaks directions volume to be saved.
+        out_peaks_values : string, optional
+            Name of the peaks values volume to be saved.
+        out_peaks_indices : string, optional
+            Name of the peaks indices volume to be saved.
+        out_gfa : string, optional
+            Name of the generalized FA volume to be saved.
```
Comment (skoudoro) [inline dipy/workflows/reconst.py:335]: the order of the docstring should align/match with the order of function arguments.

Can you update that? (`extract_pam_values`, `parallel`,  `num_processes` should be before `out_dir`) and thank you

File: dipy/workflows/reconst.py [hunk 2]
```diff
@@ -916,6 +976,28 @@ def run(self, input_files, bvalues_files, bvectors_files, mask_files,
...
```
Comment (skoudoro) [inline dipy/workflows/reconst.py:1000]: the order of the docstring should align/match with the order of function arguments.

Can you update that? (`extract_pam_values`, `parallel`,  `num_processes` should be before `out_dir`) and thank you

**Review by skoudoro — COMMENTED** [review]
Hi @alexrockhill,

Thank you for this and sorry for the late review. 

Overall, it looks good to me. See below my comment and then it should be ready to be merged

**Comment by skoudoro** [comment]
I will go ahead and merge this PR. 

Concerning the docstring fix, I will do it on a follow up PR that will update more stuff also.

Thank you for this @alexrockhill,

merging

**Comment by alexrockhill** [comment]
Thanks, sorry went on vacation after marking ready, thanks in advance for fixing the docstring.

**Comment by skoudoro** [comment]
> Thanks, sorry went on vacation after marking ready, thanks in advance for fixing the docstring.

No worries and thanks for this work!

**Comment by jhlegarreta** [comment]
@alexrockhill I'd dare to say that this PR introduced new warnings to DIPY:
https://github.com/dipy/dipy/actions/runs/8485011666/job/23305566306#step:10:4489

Before that PR we were down to 2 warnings:
https://github.com/dipy/dipy/actions/runs/8540531326/job/23403659974#step:10:4382

We are trying to remove all warnings so that we can soon switch to making warnings be reported as errors in the CIs and thus avoid introducing new warnings.

Can you please contribute with a PR to remove those warnings? Thanks

**Comment by alexrockhill** [comment]
Sure, no problem

**Pull request merged**